src/SimpleTraits.jl

```diff
@@ -320,16 +320,16 @@ let
         retsym = gensym()
         if hasmac
             fn = if oldfn_syntax
-                :(@dummy $fname{$(typs...)}($val, $(strip_kw(args1)...); $(kwargs...)) = ($pushloc; $retsym = $fbody; $poploc; $retsym))
+                :(Base.@__doc__ @dummy $fname{$(typs...)}($val, $(strip_kw(args1)...); $(kwargs...)) = ($pushloc; $retsym = $fbody; $poploc; $retsym))
             else
-                :(@dummy $fname($val, $(strip_kw(args1)...); $(kwargs...)) where {$(typs...)} = ($pushloc; $retsym = $fbody; $poploc; $retsym))
+                :(Base.@__doc__ @dummy $fname($val, $(strip_kw(args1)...); $(kwargs...)) where {$(typs...)} = ($pushloc; $retsym = $fbody; $poploc; $retsym))
             end
-            fn.args[1] = mac # replace @dummy
+            fn.args[findfirst(ex->isexpr(ex) && ex.head==:macrocall, fn.args)].args[1] = mac # replace @dummy
```

Comment:
I think this line does not work anymore.

Reply:
Mh, why not? Works fine with julia v1.1 for me. The `findfirst` call should return 3 in any case I am aware of. First element in `fn.args` is the symbol `__doc__` macro, second a `LineNumberNode` and the third the actual function macrocall (see the following exerpt from the ast). I didn't hardcode the 3 just to be safe.

```
3: Expr
      head: Symbol macrocall
      args: Array{Any}((3,))
        1: Symbol @dummy
        2: LineNumberNode
          line: Int64 325
          file: Symbol /home/tehrengruber/.julia/dev/SimpleTraits/src/SimpleTraits.jl
        3: Expr
          head: Symbol =
          args: Array{Any}((2,))
            1: Expr
              head: Symbol where
              args: Array{Any}((2,))
                1: Expr
                2: Symbol K
            2: Expr
              head: Symbol block
              args: Array{Any}((8,))
                1: LineNumberNode
                2: Nothing nothing
                3: LineNumberNode
                4: Expr
                5: LineNumberNode
                6: Nothing nothing
                7: LineNumberNode
                8: Symbol ##376
```

Reply:
Ah, sorry, I was one build behind and was wondering why it didn't work.